### ISSUE TITLE
Invalid `collapsible_match` lint.

### ISSUE THREAD

**Issue by msix29** [rust-lang/rust-clippy#13326]
### Summary

The `collapsible_match` lint is shown in a case where doing it is experimental (thus causes errors).

### Reproducer

I tried this code:

```rust
match event {
    // other irrelevant match arms.

    Event::Loop(loop_event) => {
        if let Loop::Render(render_args) = loop_event {
            app.render(&render_args)
        }
    }
    _ => (),
}
```

I expected to see this happen: No lint to be shown.

Instead, this happened: A lint was shown at the `if let Loop` part.

### Version

```text
rustc 1.80.1 (3f5fd8dd4 2024-08-06)
binary: rustc
commit-hash: 3f5fd8dd41153bc5fdca9427e9e05be2c767ba23
commit-date: 2024-08-06
host: x86_64-pc-windows-gnu
release: 1.80.1
LLVM version: 18.1.7
```

**Comment by Jarcho** [comment]
What stops `Event::Loop(Loop::Render(render_args)) => { ... }` from working? Note if the wildcard pattern changes to do something the lint wont trigger.

**Comment by msix29** [comment]
As I said at the very start of the post - "... where doing it is experimental (thus causes errors)." I'll attach images in a second.

**Comment by msix29** [comment]
![image](https://github.com/user-attachments/assets/31f56b63-bedd-4b1d-b6c1-44d029405102)
![image](https://github.com/user-attachments/assets/b2962c66-0edf-439a-af2e-2c1ed9c8615d)

**Comment by y21** [comment]
The lint is suggesting
```rs
match event {
    // other irrelevant match arms.

    Event::Loop(Loop::Render(render_args)) => {
        app.render(&render_args)
    }
    _ => (),
}
```
You should remove the `if let` entirely as the if let pattern is inlined into the match pattern.

**Comment by msix29** [comment]
Oh, that would certainly make sense. I have not encountered that lint suggesting this edit before, so I didn't comprehend it. Don't you think it would be preferable to clarify it? Or perhaps, instead of having the `try ...` part in the diagnostic (which is very unreadable), you could include a code action? I apologize for the inconvenience. 😅
